sstables/sstables_manager.cc

```diff
@@ -74,15 +74,15 @@ future<> storage_manager::stop() {
 }
 
 future<> storage_manager::update_config(const db::config& cfg) {
-    for (auto [ep, ecfg] : cfg.object_storage_config()) {
-        auto s3_cfg = make_lw_shared<s3::endpoint_config>(std::move(ecfg));
-        auto [it, added] = _s3_endpoints.try_emplace(ep, std::move(s3_cfg));
-        if (!added) {
-            if (it->second.client != nullptr) {
-                co_await it->second.client->update_config(s3_cfg);
-            }
-            it->second.cfg = std::move(s3_cfg);
+    _s3_endpoints.clear();
+
+    for (auto& e : cfg.object_storage_endpoints()) {
+        auto s3_cfg = make_lw_shared<s3::endpoint_config>(std::move(e.config));
+        auto [it, _] = _s3_endpoints.emplace(e.endpoint, std::move(s3_cfg));
+        if (it->second.client != nullptr) {
```

Comment:
How can this happen? The _s3_endpoints is cleared before population, so emplace should always return iterator to client-less entry